### PR TITLE
Update jmh configuration for async profiler

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
We don't have to use sbt-jmh-extras because the integration with async-profiler and JFR are merged to the JMH. https://github.com/sbt/sbt-jmh#using-java-flight-recorder--async-profiler

Modification:
- Remove `sbt-jmh-extras` dependency
- Update README.md and build.gradle

Result:
- Correct guide for JMH async profiler

### CODE REVIEW

File: benchmarks/jmh/README.md [hunk 1]
```diff
@@ -42,33 +33,35 @@ $ ./gradlew --no-daemon :benchmarks:jmh:clean :benchmarks:jmh:jmh ...
...
+    $ ./gradlew :benchmarks:jmh:jmh \
+      "-Pjmh.profilers=async:libPath=$HOME/async-profiler/build/lib/libasyncProfiler.dylib;output=flamegraph;dir=$HOME/result"
+    ```
 
 ## Notes
 
 - Do not forget to wrap `-Pjmh.params` and `-Pjmh.profilers` option with double quotes, because otherwise your
   shell will interpret `;` as the end of the command.
-- See [sbt-jmh documentation](https://github.com/ktoso/sbt-jmh#using-async-profiler) for more profiler options.
+- Run `$ ./gradlew :benchmarks:jmh:jmh "-Pjmh.profilers=async"` for more profiler options.
```
Comment (jrhee17) [inline benchmarks/jmh/README.md:67]: Running this actually threw an exception for me 😅 
```bash
Profilers failed to initialize, exiting.
Unable to load async-profiler. Ensure asyncProfiler library is on LD_LIBRARY_PATH (Linux), DYLD_LIBRARY_PATH (Mac OS), or -Djava.library.path. Alternatively, point to explicit library location with -prof async:libPath=<path>.
```

**Review by jrhee17 — APPROVED** [review]
Left some minor comments, but looks good 👍   Verified the commands for macos from my machine as well. Thanks @minwoox 🙇 👍 🙇

File: benchmarks/jmh/README.md (hunk 1 above)
Comment (minwoox) [reply benchmarks/jmh/README.md:67]: Oops, fixed, thanks! 😓

**Review by ikhoon — APPROVED** [review]
LGTM!
Sorry for the late review. 🙇‍♂️

**Pull request merged**